### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by CenterforCyberIntelligence**
Minor updates to README documentation - point to new GitHub REST API Documentation.

### CODE REVIEW

**Comment by s-t-e-v-e-n-k** [comment]
I'm not convinced the API is the same between the GitHub API and the Enterprise API -- I'm having a deeper think about what to do about the massive docs reorg GitHub has done.

**Comment by CenterforCyberIntelligence** [comment]
@s-t-e-v-e-n-k - I can't say that I am super familiar with historic GitHub API documentation as I am relatively new to using the API. However, when attempting to visit the API documentation linked in the README documentation here, GitHub redirects you to the link I provided in the Pull Request - just something I thought was worth checking out 👍

File: README.md [hunk 1]
```diff
@@ -9,11 +9,10 @@
 [![codecov](https://codecov.io/gh/PyGithub/PyGithub/branch/master/graph/badge.svg)](https://codecov.io/gh/PyGithub/PyGithub)
 [![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
 
-PyGitHub is a Python library to access the [GitHub API v3] and [Github Enterprise API v3].
+PyGitHub is a Python library to access the [GitHub REST API].
 This library enables you to manage [GitHub] resources such as repositories, user profiles, and organizations in your Python applications.
 
-[GitHub API v3]: https://developer.github.com/v3
-[Github Enterprise API v3]: https://developer.github.com/enterprise/v3/
+[GitHub REST API]: https://docs.github.com/en/free-pro-team@latest/rest
```
Comment (sfdye) [inline README.md:15]: I think `https://docs.github.com/rest` is more intuitive, it's just a 301
```suggestion
[GitHub REST API]: https://docs.github.com/rest
```

Comment (CenterforCyberIntelligence) [reply README.md:15]: Your suggested URL redirects to https://docs.github.com/en/free-pro-team@latest/rest - it is the same URL that I proposed in the original Pull request.

**Comment by s-t-e-v-e-n-k** [comment]
Closing in favour of #1896, which incorporates this change, thanks for your contribution!

**Pull request closed without merging**